### ISSUE TITLE
Expect user 'gitea' but current user is: 

### ISSUE THREAD

**Issue by tinxx** [go-gitea/gitea#4894]
- Gitea version (or commit ref): 1.5.1
- Git version: 2.18.0
- Operating system: Alpine Linux v3.8
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist: --

## Description
Gitea is unable to identify current user on Alpine Linux.  
I have build a simple docker container using Alpine Linux to run the official release of Gitea taken from `dl.gitea.io`. Problem is that it does not seem to be able to identify current user, so if I set a `RUN_USER` in `app.ini`, all I get is the following error message:

```
2018/09/08 12:20:59 [...s/setting/setting.go:990 NewContext()] [E] Expect user 'gitea' but current user is: 
```

Gitea runs if you omit setting `RUN_USER`, but it uses empty user name in ssh strings, too, like so: `ssh://@domain.tld/RepoName.git`

**Comment by SagePtr** [comment]
Now gitea relies on USER environment variable (don't know why, it can easily be spoofed).
So try to run
```USER=gitea /path/to/gitea web```

Maybe this should be fixed and CurrentUsername function should use Current() function from "[os/user](https://golang.org/pkg/os/user/#Current)" to determine actual username rather than looking at ```USER``` environment variable?

**Comment by adelowo** [comment]
> A follow-up problem is that ssh connection is not possible. This is due to the same problem, as gitea is called as part of the auth (ref. .ssh/authorized_keys).

`ssh -o PreferredAuthentications=password -o PubkeyAuthentication=no server@IP`

**Comment by SagePtr** [comment]
Try setting ```HOME``` environment variable as well, probably it relies on it when accessing .ssh/authorized_keys file

**Comment by tinxx** [comment]
No, I think you misunderstood. `HOME` is already set. The problem must be something Gitea executes after SSH auth is granted.

**EDIT:**
I fixed it by installing and providing `dash` as default shell for user gitea. We could still consider fixing busybox compatibility.

I think I'm fine now. Thanks for your help! :)

**EDIT 2:**
Sadly, I was mistaken. Gitea seems to rely on `bash` after. Pull worked just fine but this happens when trying to push:

```
remote: env: can't execute 'bash': No such file or directory
```

**EDIT 3:**
I replaced `dash` with `bash` in my previous fix (see above) and all seems to be good now.

I think, though, that Gitea should be made POSIX compliant so it could run on any standard shell!
